spiped/dispatch.c

```diff
@@ -80,7 +82,7 @@ doaccept(struct accept_state * A)
 	int rc = 0;
 
 	/* If we can, accept a new connection. */
-	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL)) {
+	if ((A->nconn < A->nconn_max) && (A->accept_cookie == NULL) && !(A->shutdown)) {
```

Comment:
`doaccept` seems to prepare for an incoming connection in advance, rather than choosing whether to accept a connection which is happening right now.  I would be tempted to call it "prepare_for_future_connection".

We can see this by adding a few `printf`s to the code:

```
$ ./spiped/spiped -d -s 127.0.0.1:8001 -t 127.0.0.2:8002 -k blank.txt -F
dispatch doaccept
    // wait a few seconds, then run "telnet 127.0.0.1 8001" in a different terminal
dispatch callback_gotconn
dispatch doaccept
```

In order to achieve what you want, I would start looking at `callback_gotconn` rather than `doaccept`.

That said, I'm not intimately familiar with the `spiped` code & design, and I'm not certain this is a good way to approach the problem.  I recommend waiting a few more days for @cperciva to chime in (this is a very busy time for him).